src/index.js

```diff
@@ -262,30 +262,18 @@ class SignatureFactory {
     this.method = method;
   }
 
-  getCanonicalHeaders () {
-    return {
-      date: `date:${this.dates.usbldate}`,
-      host: `host:${this.host}\n`
-    };
-  }
-
   hash (string, encoding) {
     return crypto.createHash('sha256').update(string, 'utf8').digest(encoding);
   }
 
   canonicalString () {
-    const queryStr = this.queryParameters;
-    const bodyHash = this.hash('', 'hex');
-    const canonicalHeaders = this.getCanonicalHeaders();
-
     return [
-      this.method || 'GET',
-      this.url,
-      queryStr,
-      canonicalHeaders.date,
-      canonicalHeaders.host,
-      'date;host',
-      bodyHash
+      this.method || 'GET',               // HTTPRequestMethod
+      this.url,                           // CanonicalURI
+      this.queryParameters,               // CanonicalQueryString
+      `host:${this.host}`+ '\n' + `x-usbl-date:${this.dates.longdate}`+ '\n', // CanonicalHeaders
+      'host;x-usbl-date',                 // SignedHeaders
```

Comment:
True, implementing your solution is not difficult.
But in structure, it would ease the way of problems as you disconnect the naming and the ordering.